### PR TITLE
Extracting code out of SChannel_connect_step1

### PR DESCRIPTION
**Pull request by ebejan**
Extracting code out of Schannel_connect_step1 to
1. reduce complexity
2. prepare for https://github.com/curl/curl/pull/6718

### CODE REVIEW

File: lib/vtls/schannel.c
Comment (emilengler) [inline lib/vtls/schannel.c:519]: I think this could be spitted up even more here

**Review by emilengler — CHANGES_REQUESTED** [review]

File: lib/vtls/schannel.c
Comment (ebejan) [reply lib/vtls/schannel.c:519]: The code can be split into multiple pieces, but that has no immediate benefits. This is split here, so that we can accommodate #6718.

Comment (mback2k) [reply lib/vtls/schannel.c:519]: Yeah, that could be another PR. To be able to trace structural changes in security related code, I would like to keep the moved code per PR/commit to a minimum and also avoid a mixture of just moving and actually changing code.

File: lib/vtls/schannel.c
Comment (mback2k) [inline lib/vtls/schannel.c:538]: This section (else {} block) previously contained empty lines to structure the code in logical steps, you may wanna keep those.

File: lib/vtls/schannel.c
Comment (mback2k) [inline lib/vtls/schannel.c:554]: Missing structural empty line above after move.

File: lib/vtls/schannel.c
Comment (mback2k) [inline lib/vtls/schannel.c:562]: Missing structural empty line above after move.

File: lib/vtls/schannel.c
Comment (mback2k) [inline lib/vtls/schannel.c:632]: Missing structural empty line above after move.

File: lib/vtls/schannel.c
Comment (mback2k) [inline lib/vtls/schannel.c:660]: Missing structural empty line above after move.

File: lib/vtls/schannel.c
Comment (mback2k) [inline lib/vtls/schannel.c:672]: Missing structural empty line above after move.

File: lib/vtls/schannel.c
Comment (mback2k) [inline lib/vtls/schannel.c:705]: Missing structural empty line above after move.

**Review by mback2k — CHANGES_REQUESTED** [review]
Please avoid unnecessary white space changes/removals especially if the white space is used to structure the code. Separating this chunk into its own function surely helps, but the code in the function is still a lot and therefore requires keeping some structure.

**Comment by mback2k** [comment]
You will also need to rebase your changes due to https://github.com/curl/curl/commit/ba3452cafc9920d133cb8f38ade2377be21acf0d. I would recommend to wait till the next curl release and bring this PR up-to-date for the Monday after it, so that it can be merged at the beginning of the next feature window.

**Comment by ebejan** [comment]
Looks easier to recreate the request.

**Pull request closed without merging**
